linters/psscriptanalyzer/lint.ps1

```diff
@@ -0,0 +1,33 @@
+param(
+    [Parameter(Mandatory = $true)]
+    [string]$ModuleDir,
+    [Parameter(Mandatory = $true)]
+    [string]$FilePath,
+    [Parameter(Mandatory = $true)]
+    [string]$OutputPath
+)
+
+# Import PSScriptAnalyzer module
+Import-Module -Name (Join-Path $ModuleDir PSScriptAnalyzer.psd1)
+
+# Download and import ConvertToSARIF module
+# NOTE: This could be converted to a tool once there is a way to pass in root directories of tools other than the linter itself
+$Version = "1.0.0"
+$ImportPath = Join-Path "TEMP:" "ConvertToSARIF" $Version
+Save-Module -Name ConvertToSARIF -Path "TEMP:" -RequiredVersion $Version
+Import-Module -Name (Join-Path $ImportPath "ConvertToSARIF.psd1")
```

Comment:
This is a good callout about tool dir resolution, and it's something we will look into adding in the future.

With that being said, the code as you have written here doesn't work for me for a couple reasons:
1. This doesn't work in parallel since the installations (and removal) will clobber each other
2. Not imperative, but it would be better to not have to reinstall ConvertToSarif on every execution
3. `TEMP` doesn't resolve correctly on my machine. I think we would want to use something like `$env:TEMP` and default it to `/tmp`.

We have a couple options to address this:
1. Does powershell have a convenient way to search for `ConvertToSarif.psd1` in the PATH? Tools work under the hood by just exporting the installation PATH for the linter.
2. Don't install `ConvertToSarif`, and instead use a Trunk custom parser ([example](https://github.com/trunk-io/plugins/blob/main/linters/sqlfluff/sqlfluff_to_sarif.py)). The output format looks CSV-like so that might be an option to eliminate the dependency
4. We expose the environment variable `${shared_cachedir}` for cases like this, but then the script would have to be thread-safe.

What are your thoughts on these options?

Reply:
Did not think about this needing to run in parallel, good callout. Also `TEMP:` (important to include the colon) is a PowerShell virtual drive that always points to an appropriate temp folder for the platform it is on (and is only available inside pwsh itself).

Comments on the options:
1. So I thought about trying to import `ConvertToSARIF.psd1`, but tool shims aren't actually symlinks, at least on my install, they are a bash wrapper script that then goes and points at the actual tool. Is there a way to have it be a symlink? If so, option 1 might work.
2. Certainly doable, but as it's the most work I'd prefer to try another option if I can.
3. Interesting, I wasn't aware of that variable. Can you elaborate more on what you mean by needing to make sure the script is thread safe? PowerShell is single-threaded by default.

Reply:
Thanks for the `TEMP:` explanation. I was running into some issues but it might have been a different problem. I will give it another try when we get the sarif bit figured out.

1. It isn't the shims that we would actually be leveraging. When you specify `tools: [psscriptanalyzer, pwsh]`, it implicitly adds those tools/downloads to the `PATH` when executing the linter. You can see this if you run `trunk check --verbose --force linters/psscriptanalyzer/test_data/check.in.ps1` after enabling and then inspect the yaml file. I'm not very familiar with how powershell resolves commands from the PATH, but if we can search from there we could make this a tool and Trunk will handle the installation.
2. Definitely agree.
3. It's not something we document since it's mainly an implementation detail, but it might work here. I'm not aware of the powershell options for something like this, but a naive implementation would be:

---

1. Check if installed `${shared_cachedir}/ConvertToSARIF` exists. If it does, good to execute.
2. If not, make a file `${shared_cachedir}/ConvertToSARIF.lock` and lock it.
3. Download/install the module to `${shared_cachedir}/ConvertToSARIF`
4. Unlock `${shared_cachedir}/ConvertToSARIF.lock`
5. Execute the linting

Or something like [this](https://stackoverflow.com/questions/63727114/file-lock-mechanism-using-powershell)

Reply:
Thanks for the fast and detailed explanations. I'll pursue options 1, 3, and 2 in that order, as I think all are viable.

Reply:
Thanks again for implementing this. I'm happy to help more with any minutiae, but my powershell knowledge is limited. Looking forward to getting this working!

Reply:
Option 1 turned out to be quite easy now that I know the downloads are in the PATH. I like this a lot better.